packages/flutter_tools/lib/src/project_validator.dart

```diff
@@ -21,6 +26,170 @@ abstract class ProjectValidator {
   Future<List<ProjectValidatorResult>> start(FlutterProject project);
 }
 
+/// Validator run for all platforms that extract information from the pubspec.yaml.
+///
+/// Specific info from different platforms should be written in their own ProjectValidator.
+class MachineDumpProjectValidator extends ProjectValidator{
```

Comment:
Why have a different code path for the machine dump, rather than standardizing on a single set of validators used in both --machine and non-machine modes, and simply print them differently?

Reply:
I'm not certain how we are choosing which properties to include in the non-machine --suggestions, but it seems that the machine version will expose much deeper and less-human-relevant properties. Also, the non-machine validators encode the `name` and `value` as human readable while this command prefers a direct `blah.foo.bar` structure for the name and a json compatible string as value.

Converting between the human readable version and the machine version seems tedious and unnecessary, especially since I'm not doing any "work" here, just dumping values.

Reply:
can we have --machine mode be a superset of the non-machine mode, then? I'd like to avoid having essentially two different code paths to maintain.

Reply:
The cleanest way to do this is still the separate code path. The way the validator API works is that it stores `name` - `value` pairings as two strings in each `ProjectValidatorResult` that it produces.

For example, in the existing --suggestions validator, we store the app's pubspec name as `ProjectValidatorResult(name: 'App name', value: 'my_app', status: StatusProjectValidator.success)`. There is no way of converting this to the desired machine format of `ProjectValidatorResult(name: 'FlutterProject.manifest.appName', value: '"my_app"', status: StatusProjectValidator.info)` without just hardcoding the different `name` and `value` properties. This same issue is repeated for every single existing validation. Any benefit of overlapping validations is made moot by having to write a bunch of if statements on what string to store creating much more complicated, harder to maintain, and longer code.

The other concern is that the existing validator is simply not trying to accomplish the same task as the new validator I'm adding. It is trying to check if different properties are good or not, while the dump validator is just trying to expose the values regardless of the correctness. Lumping them together seems wrong.